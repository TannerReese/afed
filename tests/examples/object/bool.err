line 18, column 16 Eval Error: Binary operator % not implemented between types boolean and boolean
line 19, column 17 Eval Error: Binary operator // not implemented between types boolean and boolean
line 20, column 15 Eval Error: Binary operator ^ not implemented between types boolean and boolean
line 27, column 11 Eval Error: Binary operator + not implemented between types ternary and ternary
line 28, column 10 Eval Error: Binary operator * not implemented between types ternary and number
5 Eval Errors encountered
