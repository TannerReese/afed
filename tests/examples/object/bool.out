# Test boolean arithmetic and operators

# Boolean arithmetic operations
-true = `false`,
true + true = `false`,
true + false = `true`,
false - true = `true`,
true * false = `false`,
true * true = `true`,

# Boolean logical operations
true || false = `true`,
false && true = `false`,
false && false || true = `true`,
false && false + true = `false`,

# Boolean unimplemented operators
true % false = `Eval Error: Binary operator % not implemented between types boolean and boolean`,
false // true = `Eval Error: Binary operator // not implemented between types boolean and boolean`,
true ^ true = `Eval Error: Binary operator ^ not implemented between types boolean and boolean`,

# Ternary
if true 3 4 = `3`,
if false (if true 1 2) (if true 3 4) + 2 = `5`,

# Ternary unimplemented operators
if + if = `Eval Error: Binary operator + not implemented between types ternary and ternary`,
if * 3 = `Eval Error: Binary operator * not implemented between types ternary and number`,

